### PR TITLE
Add Ondo Finance APIs to fetch addresses

### PR DESCRIPTION
**Pull request by sirlittle**
Allowing Defi Llama to fetch all of Ondo Finances addresses in order to do its TVL calculation. This change should not change how these calculations are made, but instead should just allow Ondo Finance to add additional multisigs and tokens without needing to make a PR every time.

### CODE REVIEW

File: projects/ondofinance/index.js [hunk 1]
```diff
@@ -84,30 +95,39 @@ async function tvl(timestamp, block, chainBlocks) {
...
 
-function tvlForAllPair(allPairVault) {
-    return async (timestamp, block) => {
+function tvlForAllPairs(timestamp, block, chainBlocks) {
+    let ondoAllPairVaults = [oldAllPairVault, newAllPairVault]
+    try {
+        const data = await (await fetch("https://data.ondo.finance/v1/addresses")).json()
+        ondoAllPairVaults = data["ondo_all_pair_vaults"]
+    } catch {
+        console.log("Failed to data from Ondo Finance, resorting to defaults")
```
Comment (g1nt0ki) [inline projects/ondofinance/index.js:104]: same comment as above, please throw an error here.

File: projects/ondofinance/index.js [hunk 2]
```diff
@@ -61,18 +61,29 @@ async function addEthBalances(addresses, block, balances) {
...
+    let partner_tokens = STABLE_PARTNER_TOKENS
+    let ondo_multisigs = STABLE_PARTNER_VAULTS
+    let ondo_lps = PARTNER_LPS
+    try {
+        const data = await (await fetch("https://data.ondo.finance/v1/addresses")).json()
+        partner_tokens = data["supported_tokens"]
+        ondo_multisigs = data["ondo_multisigs"]
+        ondo_lps = data["ondo_lps"]
+    } catch {
+        console.log("Failed to data from Ondo Finance, resorting to defaults")
```
Comment (g1nt0ki) [inline projects/ondofinance/index.js:73]: Hi, the change looks good. But we would like to know when there is an issue, and much prefer you throwing an error and failing. Another reason is, now the chart would be inconsistent, api fail leads to lower tvl right?

**Review by g1nt0ki — CHANGES_REQUESTED** [review]

**Comment by llamatester** [comment]
Error while running adapter at projects/ondofinance: 
        
 
  
    
    
    Error: Protocol doesn't have total tvl, make sure to export a tvl key either on the main object or in one of the chains
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:206:11
        at processTicksAndRejections (node:internal/process/task_queues:96:5)

File: projects/ondofinance/index.js (hunk 2 above)
Comment (sirlittle) [reply projects/ondofinance/index.js:73]: Fair, in that case, the try catch is probably unnecessary all together, will go ahead and remove.

**Comment by llamatester** [comment]
The adapter at projects/ondofinance exports TVL: 
        
 
 
    ethereum                  121.45 M
    
    total                    121.45 M

File: package-lock.json [hunk 3]
```diff
@@ -16604,7 +16604,7 @@
         "graphql": "^15.5.0",
         "graphql-request": "^3.4.0",
         "lodash": "^4.17.21",
-        "node-fetch": "2.6.7"
+        "node-fetch": "^2.6.1"
```
Comment (g1nt0ki) [inline package-lock.json:16607]: can you please revert the changes made to this file?

File: projects/ondofinance/index.js [hunk 4]
```diff
@@ -61,18 +26,23 @@ async function addEthBalances(addresses, block, balances) {
 
 async function tvl(timestamp, block, chainBlocks) {
     const balances = {};
+    const info = await fetch("https://data.ondo.finance/v1/addresses")
```
Comment (g1nt0ki) [inline projects/ondofinance/index.js:29]: changes to this file look good

File: package-lock.json (hunk 3 above)
Comment (sirlittle) [reply package-lock.json]: reverted those changes, PR should be ready to merge

**Comment by llamatester** [comment]
The adapter at projects/ondofinance exports TVL: 
        
 
 
    ethereum                  122.50 M
    
    total                    122.50 M
